vendor/github.com/openshift/library-go/pkg/proc/reaper.go

```diff
@@ -3,13 +3,87 @@
 package proc
 
 import (
+	"bufio"
+	"io/ioutil"
 	"os"
 	"os/signal"
+	"path/filepath"
+	"strconv"
+	"strings"
 	"syscall"
+	"time"
 
 	"k8s.io/klog"
 )
 
+// parseProcForZombies parses the current procfs mounted at /proc
+// to find proccess in the zombie state.
+func parseProcForZombies() ([]int, error) {
+	files, err := ioutil.ReadDir("/proc")
+	if err != nil {
+		return nil, err
+	}
+
+	var zombies []int
+	for _, file := range files {
+		processID, err := strconv.Atoi(file.Name())
+		if err != nil {
+			break
+		}
+		stateFilePath := filepath.Join("/proc", file.Name(), "status")
+		fd, err := os.Open(stateFilePath)
+		if err != nil {
+			klog.V(4).Infof("Failed to open %q: %v", stateFilePath, err)
+			continue
+		}
+		defer fd.Close()
+		fs := bufio.NewScanner(fd)
+		for fs.Scan() {
+			line := fs.Text()
+			if strings.HasPrefix(line, "State:") {
+				if strings.Contains(line, "zombie") {
+					zombies = append(zombies, processID)
+				}
+				break
+			}
+		}
+	}
+
+	return zombies, nil
+}
+
+// StartPeriodicReaper starts a goroutine to reap processes periodically if called
+// from a pid 1 process.
+// The zombie processes are reaped at the beginning of next cycle, so os.Exec calls
```

Comment:
This comment isn't necessary, since os.Exec doesn't care about zombies (it will never itself have a zombie because it never quit so the child isn't reparented)